### ISSUE TITLE
File Decrypting but not working

### ISSUE THREAD

**Issue by abferris** [apprenticeharper/DeDRM_tools#1130]
Windows 10
Downloaded from Amazon website with a kindle selected.
I have the serial number of the kindle selected copied and put into elnk Kindle Books.
It looks like it's not finding the kindle number 

calibre Debug log
calibre 4.19  embedded-python: True is64bit: False
Windows-10-10.0.18362-SP0 Windows ('32bit', 'WindowsPE')
32bit process running on 64bit windows
('Windows', '10', '10.0.18362')
Python 2.7.16+
Windows: ('10', '10.0.18362', 'SP0', u'Multiprocessor Free')
Interface language: None
Successfully initialized third party plugins: DeDRM (6, 8, 0) && KePub Output (3, 2, 11) && Manage Series (1, 2, 9)
calibre 4.19  embedded-python: True is64bit: False
Windows-10-10.0.18362-SP0 Windows ('32bit', 'WindowsPE')
32bit process running on 64bit windows
('Windows', '10', '10.0.18362')
Python 2.7.16+
Windows: ('10', '10.0.18362', 'SP0', u'Multiprocessor Free')
Interface language: None
Successfully initialized third party plugins: DeDRM (6, 8, 0) && KePub Output (3, 2, 11) && Manage Series (1, 2, 9)
Turning on automatic hidpi scaling
devicePixelRatio: 1.0
logicalDpi: 96.0 x 96.0
physicalDpi: 81.5518394649 x 81.6428571429
Using calibre Qt style: True
[0.00] Starting up...
[0.03] Showing splash screen...
[7.80] splash screen shown
[7.80] Initializing db...
[7.88] db initialized
[7.88] Constructing main UI...
[11.22] main UI initialized...
[11.22] Hiding splash screen
[18.27] splash screen hidden
[18.27] Started up in 18.27 seconds with 441 books
stdout+stderr from file dialog helper: ['', '']
piped data from file dialog helper: ['\x84\x86h\x7f{\x15\xc7W\r\xd1c\x03\x08\xcb\x1f`\x1ak\x82\x98Q&\rZgq\xe2\x06\x8a\xdf\xb4\xb7', 'C:\\Users\\gabbi\\Documents\\E-Books\\AZW FORMAT\\Under the Dome_ A Novel.azw3']
DeDRM v6.8.0: Trying to decrypt Under the Dome_ A Novel.azw3
Using Library AlfCrypto DLL/DYLIB/SO
MobiDeDrm v0.42.
Copyright © 2008-2017 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: Under the Dome: A Novel
Found 6 keys to try after 0.3 seconds
Crypto Type is: 2
DeDRM v6.8.0: Failed to decrypt with error: No key found in 6 keys tried.
DeDRM v6.8.0: Looking for new default Kindle Key after 0.3 seconds
searching for kinfoFiles in C:\Users\gabbi\AppData\Local
Found K4PC 1.25+ kinf2018 file: C:\Users\gabbi\AppData\Local\Amazon\Kindle\storage\.kinf2018
Decrypted key file using IDString '270953227' and UserName '6761626269'
DeDRM v6.8.0: Found 1 new key
MobiDeDrm v0.42.
Copyright © 2008-2017 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: Under the Dome: A Novel
Got DSN key from database default_key_1
Found 4 keys to try after 10.3 seconds
Crypto Type is: 2
DeDRM v6.8.0: Ultimately failed to decrypt after 10.3 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 172, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 625, in run
  File "calibre_plugins.dedrm.__init__", line 575, in KindleMobiDecrypt
DeDRMError: DeDRM v6.8.0: Ultimately failed to decrypt after 10.3 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Added Under the Dome: A Novel to db in: 1.9
Added 1 books in 13.6 seconds

**Comment by ElleKayEm** [comment]
The key(s) you have entered aren't right for this book.  Did you choose an E-ink Kindle (not a Fire) when downloading the book?  Make sure you have its serial number entered correctly.  Try copy and paste from the website.

**Comment by abferris** [comment]
The device selected to download from Amazon was a Kindle Paperwhite. We have the serial number copied from the website, but it's still not working. We do also have a Kindle Fire, if there is an easier configuration for that one. DRM is applied on download, not purchase, right?

**Comment by ElleKayEm** [comment]
No, you cannot decrypt a book for a Fire.  If the device chosen was the Paperwhite and you have its serial in there correctly, it should work.  Only other thing I'd suggest is using 64-bit calibre.

**Comment by abferris** [comment]
I downloaded the 64-bit Calibre. Then uploaded another Kindle book into Calibre, but when I try to convert to a different format, it still gives me the DRM message. Here is the debug log. Any other suggestions?

calibre Debug log
calibre 4.19 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.18362-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.18362')
Python 2.7.16+
Windows: ('10', '10.0.18362', 'SP0', u'Multiprocessor Free')
Interface language: None
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (1, 32, 0) && DeDRM (6, 8, 0) && Package KFX (from KFX Input) (1, 32, 0) && KFX metadata reader (from KFX Input) (1, 32, 0) && KFX Input (1, 32, 0) && KePub Output (3, 2, 12) && Manage Series (1, 2, 9)
calibre 4.19 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.18362-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.18362')
Python 2.7.16+
Windows: ('10', '10.0.18362', 'SP0', u'Multiprocessor Free')
Interface language: None
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (1, 32, 0) && DeDRM (6, 8, 0) && Package KFX (from KFX Input) (1, 32, 0) && KFX metadata reader (from KFX Input) (1, 32, 0) && KFX Input (1, 32, 0) && KePub Output (3, 2, 12) && Manage Series (1, 2, 9)
Turning on automatic hidpi scaling
devicePixelRatio: 1.0
logicalDpi: 96.0 x 96.0
physicalDpi: 81.5518394649 x 81.6428571429
Using calibre Qt style: True
[0.00] Starting up...
[0.00] Showing splash screen...
[0.70] splash screen shown
[0.70] Initializing db...
[0.72] db initialized
[0.72] Constructing main UI...
[2.45] main UI initialized...
[2.45] Hiding splash screen
[6.88] splash screen hidden
[6.88] Started up in 6.88 seconds with 439 books
DeDRM v6.8.0: Trying to decrypt Dawn (The Xenogenesis Trilogy Book 1) (1).azw3
Using Library AlfCrypto DLL/DYLIB/SO
MobiDeDrm v0.42.
Copyright © 2008-2017 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: Dawn (The Xenogenesis Trilogy Book 1)
Found 6 keys to try after 0.1 seconds
Crypto Type is: 2
DeDRM v6.8.0: Failed to decrypt with error: No key found in 6 keys tried.
DeDRM v6.8.0: Looking for new default Kindle Key after 0.1 seconds
searching for kinfoFiles in C:\Users\gabbi\AppData\Local
Found K4PC 1.25+ kinf2018 file: C:\Users\gabbi\AppData\Local\Amazon\Kindle\storage\.kinf2018
Decrypted key file using IDString '270953227' and UserName '6761626269'
DeDRM v6.8.0: Found 1 new key
MobiDeDrm v0.42.
Copyright © 2008-2017 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: Dawn (The Xenogenesis Trilogy Book 1)
Got DSN key from database default_key_1
Found 4 keys to try after 9.5 seconds
Crypto Type is: 2
DeDRM v6.8.0: Ultimately failed to decrypt after 9.5 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 172, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 625, in run
  File "calibre_plugins.dedrm.__init__", line 575, in KindleMobiDecrypt
DeDRMError: DeDRM v6.8.0: Ultimately failed to decrypt after 9.5 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Added Dawn (The Xenogenesis Trilogy Book 1) to db in: 1.4
Added 1 books in 11.7 seconds
